### ISSUE TITLE
relic agent not forwarding logs when payload size exceed

### ISSUE THREAD

**Issue by sahilamahajan66** [newrelic/node-newrelic#1998]
Description
We are using node js relic agent to forward logs to relic server. We found case in which logs are not forwarded to new relic server as payload size exceed and we need to restart our server to enable log forwarding again.
**Completely blocks next set of logs as relic agent retaining that logs even it is not network error** 
for your reference here is code which cause issue.
This code retain logs and not clearing batch result due to which it block next batch of logs due to merging of events in event-aggregator
![image](https://github.com/newrelic/node-newrelic/assets/84334816/ae7e1ed9-0270-4ef3-8a9b-477bb3376fec)

Expected Behavior
Log should be appear in Log table except Log that has limit exceeded

Your Environment
app_name : PRODUCTION
EXPRESS: 4.18.2
apollo-server: 4.7.1
graphql: 16.6.0
Node version: 16.0.0
version: 11.5.0

**Comment by bizob2828** [comment]
@sahilmahajanpikup can you provide a reproduction case please?

**Comment by sahilmahajanpikup** [comment]
1. start logging info which has size greater than 1 MB i.e max limit
2. In specific interval some message
3. Logs keeps appending to priorityEventQueue which is not clear even max payload detected
4. it keeps reaching till default queue limit 833 and start overflow. no log forwards for logs which are in queue + overflow logs.
You can refer code below
[test.js.txt](https://github.com/newrelic/node-newrelic/files/14196610/test.js.txt)

**Comment by jsumners-nr** [comment]
Reproduction:

```js
'use strict'

const newrelic = require('newrelic')
const crypto = require('node:crypto')
const xbytes = require('xbytes')
const pino = require('pino')
const log = pino(pino.destination('/dev/null'))

newrelic.startBackgroundTransaction('log-size-exceeded', async () => {
  const allowableBytes = crypto.randomBytes(xbytes.parseSize('64KiB'))
  log.info({ bytes: allowableBytes.toString('base64') }, 'allowable log line')

  const tooManyBytes = crypto.randomBytes(xbytes.parseSize('2MiB'))
  log.info({ bytes: tooManyBytes.toString('base64') }, 'log line too big')

  log.info('!!! should still get recorded')

  newrelic.shutdown({ collectPendingData: true }, () => {})
})
```

Findings so far:

1. Logs that exceed the allowed size get found by https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/remote-method.js#L238-L249
2. That pops us over to https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/remote-method.js#L179C20-L182
3. And then https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/remote-method.js#L119-L120
4. And then https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/api.js#L470-L473
5. And then https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/api.js#L483-L488
6. And then https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/aggregators/base-aggregator.js#L99-L101
7. And then https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/aggregators/event-aggregator.js#L130-L150

The result, I am thinking at the moment, is that due to the `retainData: true` we are pushing the bad log back to the top of the stack and this prevents subsequent logs from being processed.

**Comment by bizob2828** [comment]
@jsumners-nr nice analysis.  It's clear based on the todo in https://github.com/newrelic/node-newrelic/blob/b460c7be589dd6fa801285b756bf66f62e0f10aa/lib/collector/api.js#L483-L488 that we need to figure out how to handle this by asking other language agents

**Comment by sahilamahajan66** [comment]
@bizob2828 when we can expect this to be fix?
for temporary solution what should we have to do till issue get fixed?

**Comment by bizob2828** [comment]
@sahilamahajan66 we are working on a fix.  In the meantime you can disable log forwarding in your configuration `config.application_logging.forwarding.enabled = false`

**Comment by jsumners-nr** [comment]
The resolution for this issue has been released in [v11.10.4](https://github.com/newrelic/node-newrelic/releases/tag/v11.10.4).
